pkg/kvstore/etcd.go

```diff
@@ -180,7 +180,25 @@ func newEtcdClient(config *client.Config, cfgPath string) (BackendOperations, er
 		return nil, err
 	}
 	log.Info("Waiting for etcd client to be ready")
-	s, err := concurrency.NewSession(c)
+
+	var s *concurrency.Session
+	sessionChan := make(chan *concurrency.Session)
+	errorChan := make(chan error)
+	go func() {
+		session, err := concurrency.NewSession(c)
+		sessionChan <- session
+		errorChan <- err
+		close(sessionChan)
+		close(errorChan)
+	}()
+
+	select {
+	case s = <-sessionChan:
+		err = <-errorChan
+	case <-time.After(10 * time.Second):
```

Comment:
Is 10 seconds long enough?

Reply:
10 seconds seems a bit stringent to me as well. We could make it close to, but under, the `initialDelaySeconds` in our `livenessProbe` as specified in the cilium YAML?

Reply:
I went with it because this is the default DialTimeout for etcd client. Although I think it would be better to use config.DialTimeout directly. Will change it like that, please let me know if it's ok.